src/webapp/pages/analysis/AnalysisPage.tsx

```diff
@@ -31,31 +30,59 @@ const useStyles = makeStyles(() => ({
         width: "24px",
     },
     active: {
-        backgroundColor: "#1976d2",
+        backgroundColor: "#8E8E8E",
     },
-    inactive: {
+    pending: {
         backgroundColor: "#8E8E8E",
     },
-    largeIcon: { fontSize: "32px" },
+    completed: {
+        backgroundColor: customTheme.color.intenseGreen,
+    },
+    error: {
+        backgroundColor: customTheme.color.error,
+    },
+    largeIcon: {
+        fontSize: "20px",
+        color: "#fff",
+        backgroundColor: "#1976d2",
+        borderRadius: 50,
+        padding: "0.125rem",
+    },
 }));
 
-function StepIcon(props: { completed: boolean; active: boolean; text: string }) {
-    const { active, completed, text } = props;
+function StepIcon(props: {
+    active: boolean;
+    text: string;
+    hasIssues: boolean;
+    isPending: boolean;
+}) {
+    const { active, text, hasIssues, isPending } = props;
     const classes = useStyles();
-    if (completed) return <CheckCircleIcon className={classes.largeIcon} htmlColor="#B5DFB7" />;
 
-    const mergeClasses = _([classes.circle, active ? classes.active : classes.inactive])
+    const issuesClasses = _([classes.circle, active ? classes.active : classes.error])
+        .compact()
+        .join(" ");
+    const pendingClasses = _([classes.circle, active ? classes.active : classes.pending])
+        .compact()
+        .join(" ");
+    const completedClasses = _([classes.circle, active ? classes.active : classes.completed])
         .compact()
         .join(" ");
 
-    return <div className={mergeClasses}>{text}</div>;
+    if (isPending) {
```

Comment:
Here we're returning the same component but with different className. I think it's easier to read if we have a method to calculate the classes and just return the same component.

```tsx

function StepIcon(...) {
    const buildIconClasses = React.useMemo(() => {
        const baseClasses = [classes.circle];

        if (active) {
            return [...baseClasses, classes.active].join(" ");
        } else if (isPending) {
            return [...baseClasses, classes.pending].join(" ");
        } else if (hasIssues) {
            return [...baseClasses, classes.error].join(" ");
        } else {
            return [...baseClasses, classes.completed].join(" ");
        }
    }, [classes, active, hasIssues, isPending]);

    return <div className={buildIconClasses}>{text}</div>;
}

```